sklearn/decomposition/pca.py

```diff
@@ -68,6 +69,9 @@ def _assess_dimension_(spectrum, rank, n_samples, n_features):
         v = np.sum(spectrum[rank:]) / (n_features - rank)
         pv = -np.log(v) * n_samples * (n_features - rank) / 2.
 
+    if sys.float_info.epsilon > abs(v):
```

Comment:
`sys.float_info.epsilon` means the difference between `1` and the smallest representable float greater than `1`. On my machine it's about 1e-16, and on 32-bit machines it should be about 1e-8. `np.finfo(np.float).tiny`, the smallest positive float, is much smaller. 

`pinv`'s `rcond` is 1e-15 by default:
```
    rcond : float
      Cutoff for small singular values.
      Singular values smaller (in modulus) than
      `rcond` * largest_singular_value (again, in modulus)
      are set to zero.
```